veocean-data-farming/veocean.md

```diff
@@ -0,0 +1,111 @@
+
+## Replenishing your veOCEAN
+
+You can choose to update your lock and replenish your veOCEAN balance at any time.
+
+To maximize rewards, participants would need to update their 4-year lock every week in order to maintain their veOCEAN balance as high as possible.
+
+## veOCEAN Earnings
+
+All earnings for veOCEAN holders are claimable in Ethereum mainnet. (Data assets for DF may published in any network where Ocean’s deployed in production: Eth mainnet, Polygon, etc.)
+
+There’s a new DF round every week; in line with this, there’s a new ve distribution “epoch” every week. This affects when you can first claim rewards. Specifically, if you lock OCEAN on day x, you’ll be able to claim rewards on the first ve epoch that begins after day x+7. Put another way, from the time you lock OCEAN, you must wait at least a week, and up to two weeks, to be able to claim rewards. (This behavior is inherited from veCRV. Here’s the code. )
+
+veOCEAN holders have earnings from two sources:
+
+### Earnings from Community Fees
+
+Every transaction in Ocean Market and Ocean backend generates transaction fees, some of which go to the community. 50% of the community fees will go to veOCEAN holders; the rest will go to Ocean community-oriented traction programs.
+
+All earnings here are passive.
+
+### Earnings from Data Farming
+
+veOCEAN holders will each get weekly DF rewards allocation, except a small carveout for any Data Challenge initiatives that may run through DF ops.
+
+**veOCEAN holders can be passive, though they will earn more if active.**
+
+“Being active” means allocating veOCEAN to promising data assets (data NFTs). Then, rewards follow the usual DF formula: DCV * stake. Stake is the amount of veOCEAN allocated to the data asset. There is no liquidity locked inside a datatoken pool. (And this stake is safe: you can’t lose your OCEAN as it is merely locked.)
+
+## Flow of Value
+
+The image below illustrates the flow of value. On the left, at time 0, the staker locks their OCEAN into the veOCEAN contract, and receives veOCEAN. In the middle, the staker receives OCEAN rewards every time there’s revenue to the Ocean Protocol Community (top), and also as part of Data Farming rewards (bottom). On the right, when the lock expires (e.g. 4 years) then the staker is able to move their OCEAN around again.
+
+![](./images/flow_of_value.png)
+_Flow of Value_
+
+The veOCEAN design is in accordance with the Web3 Sustainability Loop, which Ocean uses as its system-level design.
+
+The veOCEAN code was forked from the veCRV code. veCRV parameters will be the starting point. To minimize risk, tweaks will be circumspect.
+
+## Security
+
+[veOCEAN core contracts](https://github.com/oceanprotocol/contracts/tree/main/contracts/ve) use [veCRV contracts](https://curve.readthedocs.io/dao-vecrv.html) with zero changes, on purpose: the veCRV contracts have been battle-tested for two years and have not had security issues.
```

Comment:
Did the veCRV contracts also go through an audit? If so that would also be worth mentioning

Reply:
Done. Documented audit done in CRV, moved details wrt how much is locked up across ve ecosystem, and reference to crv docs for the audits.